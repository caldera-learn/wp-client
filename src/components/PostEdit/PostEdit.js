import React from 'react';
import PropTypes from 'prop-types';
import {postShape} from "../Post/postShape";

export const PostEdit = (props) =>  {

	const onChange = (update) => {
		props.onChange({
			...props.post,
			...update
		})
	};
	return(
		<div
			className={props.className}
		>
			<h3>
				<input
					type={'text'}
					value={props.post.title.rendered}
					onChange={(event) => {
						onChange({
							title: {
								rendered: event.target.value
							}
						})
					}}
				/>
			</h3>
			<div>
				{props.post.content.rendered}
			</div>
		</div>
	);
};

PostEdit.propTypes = {
	className: PropTypes.string.isRequired,
	post: postShape,
	onChange: PropTypes.func.isRequired
};

PostEdit.defaultProps = {
	className: 'post-wrapper'
};