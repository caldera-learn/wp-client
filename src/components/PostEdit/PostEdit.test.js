import React from 'react';
import {PostEdit} from "./PostEdit";
import renderer from 'react-test-renderer';
describe( 'PostEdit component', () => {
	/** Mock WordPress post to test with**/
	const post = {
		title: {
			rendered: 'Hi Roy'
		},
		content: {
			rendered: 'Lorem ipsum, etc.'
		}
	};

	const mockHandler = () => {};

	it( 'Matches snapshot with basic props', () => {
		const component = renderer.create(
			<PostEdit
				onChange={mockHandler}
				post={post}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});

	it( 'Applies className to outermost element', () => {
		const component = renderer.create(
			<PostEdit
				onChange={mockHandler}
				post={post}
				className={'food'}
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	});
});