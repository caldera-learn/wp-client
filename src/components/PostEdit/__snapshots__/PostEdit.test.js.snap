// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostEdit component Applies className to outermost element 1`] = `
<div
  className="food"
>
  <h3>
    <input
      onChange={[Function]}
      type="text"
      value="Hi Roy"
    />
  </h3>
  <div>
    Lorem ipsum, etc.
  </div>
</div>
`;

exports[`PostEdit component Matches snapshot with basic props 1`] = `
<div
  className="post-wrapper"
>
  <h3>
    <input
      onChange={[Function]}
      type="text"
      value="Hi Roy"
    />
  </h3>
  <div>
    Lorem ipsum, etc.
  </div>
</div>
`;
